src/Microsoft.ApplicationInsights.AspNetCore/Implementation/TelemetryConfigurationOptionsSetup.cs

```diff
@@ -51,9 +52,9 @@ public void Configure(TelemetryConfiguration configuration)
 
             if (this.telemetryProcessorFactories.Any())
             {
-                foreach (Func<ITelemetryProcessor, ITelemetryProcessor> processorFactory in this.telemetryProcessorFactories)
+                foreach (ITelemetryProcessorFactory processorFactory in this.telemetryProcessorFactories)
```

Comment:
@karolz-ms AspNetCore allows usage of third party DI containers if they support common basic set of features. http://structuremap.github.io/ is one of these containers.